Core/include/Acts/Definitions/Direction.hpp

```diff
@@ -27,14 +27,18 @@ class Direction final {
   };
 
  public:
-  static constexpr auto Negative = Value::Negative;
-  static constexpr auto Positive = Value::Positive;
+  static constexpr Direction Negative() { return Direction{Value::Negative}; }
+  static constexpr Direction Positive() { return Direction{Value::Positive}; }
 
-  static constexpr auto Backward = Value::Negative;
-  static constexpr auto Forward = Value::Positive;
+  static constexpr Direction Backward() { return Direction{Value::Negative}; }
+  static constexpr Direction Forward() { return Direction{Value::Positive}; }
 
-  static constexpr auto OppositeNormal = Value::Negative;
-  static constexpr auto AlongNormal = Value::Positive;
+  static constexpr Direction OppositeNormal() {
+    return Direction{Value::Negative};
+  }
+  static constexpr Direction AlongNormal() {
+    return Direction{Value::Positive};
+  }
```

Comment:
One question here is whether the static factories should start with a lower case.

Reply:
I was also thinking about this and I don't really have come to a conclusion. Having the upper case is what Eigen does and I feel like it improved readability. But it seems also be against our usual convention.

I would be good with keeping them upper case